### ISSUE TITLE
Control over the comment section

### ISSUE THREAD

**Issue by tio-trom** [friendica/friendica#13650]
I saw this post just now https://mas.to/@TechConnectify/111428308227305812 about this popular account complaining that he gets a lot of crappy comments on his posts on the fediverse and he is sick of it and wants to give up using the fediverse. 

Yes we have a lot of tools to deal with these "bad actors" that are trolling people, like we can ignore or block them, but I think a really useful feature would be to have control over the comments not just the users.

In Friendica we can already choose to only see the posts made by our friends in the Network section. Which is so relieving. Full control over the posts you see. 

However we need something similar for the comments that we are seeing on our posts. I understand that if a post is public anyone can comment from their own instances, but should not matter. What matters is for you, the owner of the post, to have control over what you see. Maybe is possible to delete unwanted comments automatically or hide them. Similarly to what it happens when you ignore/block someone and they still can comment on your posts but you do not see their comments. Neither do you see any notifications.

**Therefore I suggest the following:**

1. Disable the comments entirely for any post. (basically delete/hide them all) https://github.com/friendica/friendica/issues/7238
2. Allow your friends to comment. (delete/hide the comments from everyone else)
3. Allow friends and their friends to comment. (delete/hide the comments from everyone else)
4. Allow all of your followers and friends to comment. (delete/hide the comments from everyone else)
5. Allow friends, followers, and friends of friends to comment. (delete/hide the comments from everyone else)
6. Remove comments containing certain words.

I think it is a must to protect ourselves from the ones that can harm us.

I hope this is doable somehow.

**Comment by tio-trom** [comment]
Or more easily:

**Who can leave a comment?** 
(multiple choices)

- Friends
- Followers
- Friends of Friends
- No one
- Everyone

**Comment by haheute** [comment]
I prefer "who can leave a comment" over "who can see the comment"..

**Comment by annando** [comment]
Problem with ActivityPub is that comments don't even need to be relayed by your system. This means that when you create a public post on your system, then some user from (for example Pleroma) can comment on it in any case. And then the comments will be distributed by the Pleroma server. Also most servers will ignore any attempt from the thread owner to delete any comment that is from someone else.

**Comment by tio-trom** [comment]
> Problem with ActivityPub is that comments don't even need to be relayed by your system. This means that when you create a public post on your system, then some user from (for example Pleroma) can comment on it in any case. And then the comments will be distributed by the Pleroma server. Also most servers will ignore any attempt from the thread owner to delete any comment that is from someone else.

Yes but this is an issue with ignored/blocked users too, right? They can still comment on your posts anyway....so I was thinking to expand that for the above case I described.

**Comment by haheute** [comment]
I thought, maybe add a field like "allow_comments" on each post, and when it is "0" or "false" then no comments can be added to the post.. but its just guessing

**Comment by MrPetovan** [comment]
This setting would be deceiving. Meaning that your own node could reject incoming comments on your posts with this flag on, but they would still live on on other servers, and you could eventually find them by looking at the same conversation on a different server.

**Comment by tio-trom** [comment]
> This setting would be deceiving. Meaning that your own node could reject incoming comments on your posts with this flag on, but they would still live on on other servers, and you could eventually find them by looking at the same conversation on a different server.

I understand but isn't it the same when you ignore or block someone and they can still comment on your posts but it simply means you do not see their comments on your server but can see them via other servers?

**Comment by MrPetovan** [comment]
Indeed, but it feels to me it is a different scope from ignoring/blocking individual accounts and blocking a group of people from commenting on a single post, and as such, might create different expectations. Not that the current expectations around ignoring or blocking are 100% met...

**Comment by tio-trom** [comment]
I was surprised to find out that the ignored/blocked people can still comment on my posts is just that I do not see their comments. Took me a while to get used to it but at the end of the day it is so useful to have this option. It is for my own personal mental health. Same way that I can stop the notifications for a thread (ignore it) it gives me a peace of mind. 

Therefore having the ability to mute the comments on my posts, despite the said limitation, will have similar effects for the owners of the posts. No notifications, no visible comments. 

And maybe in the future ActivityPub will implement something that will tell every instance that these deleted comments should be deleted.

I still think such a feature to control the comments can be massively useful for all of us.
